File: app/src/main/java/com/example/papersynth/dataclasses/FourierSeries.kt
package com.example.papersynth.dataclasses

import org.jetbrains.kotlinx.multik.ndarray.data.D1
import org.jetbrains.kotlinx.multik.ndarray.data.NDArray

data class FourierSeries(
    val coefficientsA: NDArray<Float, D1>,
    val coefficientsB: NDArray<Float, D1>,
    val numTerms: Int,
    val a0: Float
) {

}

File: app/src/main/cpp/FourierSeries.cpp
//
// Created by lozbe on 6/30/2021.
//

#include "FourierSeries.h"

#include <utility>

FourierSeries::FourierSeries(
        std::vector<float> coefficientsA,
        std::vector<float> coefficientsB,
        int numTerms,
        float a0)
{
    this->coefficientsA = std::move(coefficientsA);
    this->coefficientsB = std::move(coefficientsB);
    this->numTerms = numTerms;
    this->a0 = a0;
}
